lmdeploy/model.py

```diff
@@ -1013,6 +1013,57 @@ def messages2prompt(self, messages, sequence_start=True):
         return ret
 
 
+@MODELS.register_module(name=['deepseek-chat'])
+class Deepseek(BaseModel):
+
+    def __init__(self, user='User:', assistant='Assistant:', **kwargs):
+        super().__init__(**kwargs)
+        self._user = user
+        self._assistant = assistant
+        self._bos = '<｜begin▁of▁sentence｜>'
+
+    def decorate_prompt(self, prompt, sequence_start=True):
+        """Return the prompt that is concatenated with other elements in the
+        chat template.
+
+        Args:
+            prompt (str): the input prompt
+            sequence_start (bool): indicator for the first round chat of a
+               session sequence
+        Returns:
+            str: the concatenated prompt
+        """
+        ret = f'{self._user} {prompt}\n\n{self._assistant}'
+        if sequence_start:
+            ret = f'{self._bos}{ret}'
+        return ret
+
+    def messages2prompt(self, messages, sequence_start=True):
+        """Return the prompt that is concatenated with other elements in the
+        chat template.
+
+        Only evaluate the last instruction completion pair.
+        Args:
+            messages (str | List): user's input prompt
+        Returns:
+            str: the concatenated prompt
+        """
+        if isinstance(messages, str):
+            return self.get_prompt(messages, sequence_start)
+        ret = f'{self._bos}'
+        for _, msg in enumerate(messages):
+            role, content = msg['role'], msg['content']
+            if role == self.user:
+                ret += f'{self._user} {content}\n\n{self._assistant}'
```

Comment:
![image](https://github.com/InternLM/lmdeploy/assets/41138331/b54f2753-1e39-44b0-b966-8d2863c34f89)
Tested OK with this change.